runtime-modules/project-token/src/types.rs

```diff
@@ -76,42 +137,325 @@ pub enum TransferPolicy<Hash> {
     Permissioned(Hash),
 }
 
+// TransferPolicyParams => TransferPolicy conversion
+impl<Hash, SingleDataObjectUploadParams>
+    From<TransferPolicyParams<WhitelistParams<Hash, SingleDataObjectUploadParams>>>
+    for TransferPolicy<Hash>
+{
+    fn from(
+        params: TransferPolicyParams<WhitelistParams<Hash, SingleDataObjectUploadParams>>,
+    ) -> Self {
+        match params {
+            TransferPolicyParams::Permissioned(whitelist_params) => {
+                Self::Permissioned(whitelist_params.commitment)
+            }
+            TransferPolicyParams::Permissionless => Self::Permissionless,
+        }
+    }
+}
+
 impl<Hash> Default for TransferPolicy<Hash> {
     fn default() -> Self {
         TransferPolicy::<Hash>::Permissionless
     }
 }
 
-/// The possible offering variants: This is a stub
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Debug, Default)]
+pub struct VestingScheduleParams<BlockNumber> {
+    // Vesting duration
+    pub(crate) duration: BlockNumber,
```

Comment:
This notion of duration is unclear to me, is it counting the linear part? if so, lets rename.

Reply:
Yes, will do

Reply:
Renamed to `linear_vesting_duration` in https://github.com/Joystream/joystream/pull/3618/commits/d8e70c04967c6c8050a12b204be1f39974e955ef